ledger/eval.go

```diff
@@ -676,21 +677,24 @@ func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxnWi
 			continue
 		}
 
-		// Initialize group without ApplyData lazily
+		// Initialize side effects and group without ApplyData lazily
 		if groupNoAD == nil {
 			groupNoAD = make([]transactions.SignedTxn, len(txgroup))
+			pastSideEffects = make([]*logic.EvalSideEffects, len(txgroup))
```

Comment:
```suggestion
			pastSideEffects = logic.MakePastSideEffects(len(txgroup))
```
no need ledger to know implementation details of PastSideEffects.